packages/account-template/src/index.ts

```diff
@@ -0,0 +1,131 @@
+import type { CeramicApi, CeramicDocument } from './types'
+
+import DIDDocument from './did'
+import AccountLinks from './accountLinks'
+
+
+
+// All of these could likely inherit from a AccountSubTile abstract parent class or a builder class
+
+class Keychain {
+  constructor (public ceramicDoc: CeramicDocument, private _ceramic: CeramicApi) { }
+
+  static async build(owner: string, ceramic: CeramicApi): Promise<Keychain> {
+    const genesisContent: Record<string, string[]> = {
+      "privacy-policy-read-keys": [],
+      "auth-public-keys": [],
+      "auth-data": [],
+      "legacy-data": []
+    }
+    const ceramicDoc = await ceramic.createDocument(genesisContent, 'tile', { owners: [owner] })
+    return new Keychain(ceramicDoc, ceramic)
+  }
+}
+
+class Profile {
+  constructor (public ceramicDoc: CeramicDocument, private _ceramic: CeramicApi) { }
+
+  static async build(owner: string, ceramic: CeramicApi): Promise<Profile> {
+    const genesisContent: Record<string, any> = {}
+    const ceramicDoc = await ceramic.createDocument(genesisContent, 'tile', { owners: [owner] })
+    return new Profile(ceramicDoc, ceramic)
+  }
+}
+
+class Sources {
+  constructor (public ceramicDoc: CeramicDocument, private _ceramic: CeramicApi) { }
+
+  static async build(owner: string, ceramic: CeramicApi): Promise<Sources> {
+    const genesisContent: string[] = []
+    const ceramicDoc = await ceramic.createDocument(genesisContent, 'tile', { owners: [owner] })
+    return new Sources(ceramicDoc, ceramic)
+  }
+}
+
+class Services {
+  constructor (public ceramicDoc: CeramicDocument, private _ceramic: CeramicApi) { }
+
+  static async build(owner: string, ceramic: CeramicApi): Promise<Services> {
+    const genesisContent: any[] = []
+    const ceramicDoc = await ceramic.createDocument(genesisContent, 'tile', { owners: [owner] })
+    return new Services(ceramicDoc, ceramic)
+  }
+}
+
+
+class ThreeIDAccount {
+  [key: string]: any // allow access to properties using index notation (ex: account['services'])
+
+  constructor (public ceramicDoc: CeramicDocument, public didDocument: DIDDocument, public accountLinks: AccountLinks, public keychain: Keychain, public profile: Profile, public sources: Sources, public services: Services, private _ceramic: CeramicApi, public provider?: any) { }
+
+  get DID (): string {
+    return this.didDocument.did
+  }
+
+  static async build (did: string, ceramic: CeramicApi, provider?: any): Promise<ThreeIDAccount> {
```

Comment:
Provider should not be set here. Instead it should be passed when adding a new link as described in a comment above.